File: Lab2.1cpp/Row.cpp
// Row.cpp
#include "Row.h"

Row::Row() : value("") {
}

Row::Row(const std::string& value) : value(value) {
}

Row::Row(const Row& other) : value(other.value) {
}

Row::~Row() {
}

std::string Row::getValue() const {
    return value;
}

int Row::calculateLength() const {
    return value.length();
}

std::string Row::toString() const {
    return "Row: " + value + ", Length: " + std::to_string(calculateLength());
}

File: Lab2.1/Row.cs
// Row.cs
using System;

namespace Lab3
{
    public class Row
    {
        protected string value;

        public string Value
        {
            get { return value; }
        }

        public Row() : this("") { }

        public Row(string value)
        {
            this.value = value;
        }

        public Row(Row other)
        {
            this.value = other.value;
        }

        ~Row()
        {
            Console.WriteLine("Row object is being deleted");
        }

        public int CalculateLength()
        {
            return value.Length;
        }

        public override string ToString()
        {
            return $"Row: {value}, Length: {CalculateLength()}";
        }
    }
}
